haystack/dataclasses/streaming_chunk.py

```diff
@@ -29,6 +29,22 @@ class StreamingChunk:
 StreamingCallbackT = Union[SyncStreamingCallbackT, AsyncStreamingCallbackT]
 
 
+@overload
+def select_streaming_callback(
+    init_callback: Optional[StreamingCallbackT],
+    runtime_callback: Optional[SyncStreamingCallbackT],
+    requires_async: Literal[False],
+) -> Optional[SyncStreamingCallbackT]: ...
+
+
+@overload
+def select_streaming_callback(
+    init_callback: Optional[StreamingCallbackT],
+    runtime_callback: Optional[AsyncStreamingCallbackT],
+    requires_async: Literal[True],
+) -> Optional[AsyncStreamingCallbackT]: ...
```

Comment:
@anakin87 I anticipate this might break our CI for integrations that use this method.

Reply:
Why? The original signature is still valid...

Reply:
Ahh right good point